api/control/v1alpha1/types.go

```diff
@@ -20,6 +20,25 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// TemplateExpansionAnnotationKey, when paired with the value "true" in an annotation of
+// a workload object in a WDS, indicates that Go template expansion should be
+// bundled with propagation from core to WEC.
+//
+// Go template expansion means to (1) parse each leaf string of the object as a Go template
+// as defined in the Go standard package "text/template", and (2) for each WEC, replace that
+// leaf string with the string that results from expanding this template
+// (`Template.Execute`) using data about the WEC. This data is a map from parameter
+// name to parameter value.
+//
+// A destination is described by an inventory object in an Inventory and Transport Space,
+// and its labels and annotations provide parameter names and values (with labels taking priority over annotations).
+//
+// Note that this sort of customization has limited applicability.  It can only be used where
+// the un-expanded string passes the validation conditions of the relevant object type.
+// For more broadly applicable customization, see Customizer objects.
+
+const TemplateExpansionAnnotationKey string = "control.kubestellar.io/expand-templates"
```

Comment:
can we name this annotation with a more "user-friendly" name?
I would expect something like `control.kubestellar.io/customization: true`. 
customization is a feature of KS which will be described in KS docs (both website and git readme) up completion. 
I think users of the KS may understand this much better with this name after they read the docs.

Reply:
I am open to other names. But template expansion is not the only form of customization that we contemplate, so I think that the name here should be more specific than simply "customization".

Reply:
we should specify in our docs that KS customization is based on go templating. then we should be able to use just `customization` as the annotation name. whoever wants to customize an object should read the docs and understand it's based on go templating (sounds very reasonable to me).
we should do any customization that go templating does, not necessarily string substitution. 
I think you're trying to be too specific.

Reply:
I am remembering our other conversations about other styles of customization. I call one of them "replace-this-with-that". In Karmada, they have such a concept and call it "overrides". As noted in this PR, template expansion is very limited in its expressiveness. I think it is reasonably likely that we will introduce one or more additional forms of customization in the future. That is why I do not want to equate "customization" with "template expansion".

Reply:
so in your vision, let's assume in the future we introduce one more form of customization (let's call it XXX).
would it be valid to define in a single object two types of customization? both string substitution (for some fields) and XXX (for other fields)?
sounds to me like a valid scenario, isn't it?
so now do you expect users to put multiple annotations on object? one for every type of customization?

to make a long story short - we're not there yet. and I have some doubts we will be there in the near future.
until that happens, I think documenting that string subst is the only customization option and just use customization as annotation name is much easier for understanding. 
trying to be too specific for future maybes might get very confusing. 
for example - even though I know exactly what we're doing, this name confuse me.

Reply:
For example: in release-0.14 and earlier of KubeStellar, you can see in https://github.com/kubestellar/kubestellar/blob/v0.14.0/pkg/apis/edge/v2alpha1/customize.go that we had two forms of customization and they did involve two annotations but the value of one of them was not true/false but rather a reference to an object.